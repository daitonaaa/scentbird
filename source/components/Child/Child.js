import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { connect } from 'react-redux';

import './Child.scss';

import {
  setCheckChildStatus,
  deleteItemChildAndSetCount,
} from 'actions/items';


class Child extends Component {

  static propTypes = {
    id: PropTypes.number.isRequired,
    check: PropTypes.bool.isRequired,
    title: PropTypes.string.isRequired,
    parentId: PropTypes.number.isRequired,

    setCheckChildStatus: PropTypes.func.isRequired,
    deleteItemChildAndSetCount: PropTypes.func.isRequired,
  };

  handleToggleCheck = () => {
    const {
      id,
      check,
      parentId,
      setCheckChildStatus,
    } = this.props;

    setCheckChildStatus(parentId, id, !check);
  }

  handleDeleteItemChild = () => {
    const {
      id,
      parentId,
      deleteItemChildAndSetCount,
    } = this.props;

    deleteItemChildAndSetCount(parentId, id);
  }

  render() {
    const {
      id,
      title,
      check,
    } = this.props;

    return (
      <div className={classNames('child', { check })}>
        <div className="child__title">
          <span>{id}:</span> {title}
        </div>
        <div
          onClick={this.handleToggleCheck}
          className={classNames('child__check-btn', { check })}
        >
          {check ? 'Снять выделение' : 'Выделить'}
        </div>
        <div
          className="child__delete-btn"
          onClick={this.handleDeleteItemChild}
        >
          Удалить
        </div>
      </div>
    );
  }
}


const mapDispatchToProps = {
  deleteItemChildAndSetCount,
  setCheckChildStatus,
};

export default connect(
  ()=>({}),
  mapDispatchToProps
)(Child);
