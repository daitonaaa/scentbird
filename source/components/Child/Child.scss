.child {
  display: flex;
  margin: 5px 10px 0;
  background: #fdfaff;
  font-size: 13px;
  border: solid 1px #edd7fb;
  justify-content: space-between;
  align-items: center;

  &:last-child {
    margin: 5px 10px 30px;
  }

  &.check {
    background: #647da2;
    color: #fff;
    border: solid 1px #647da2;
  }

  &__title, &__check-btn, &__delete-btn {
    padding: 5px 10px;
    transition: .2s;
  }

  &__check-btn {
    width: 135px;
    background: #a18aaf;
    color: #fff;
    text-align: center;
    cursor: pointer;

    &:hover {
      background: #baa6c5;
    }

    &.check {
      background: #fdfaff;
      color: #647da2;
    }
  }

  &__delete-btn {
    padding: 5px;
    background: #de9f9f;
    color: #fff;
    text-align: center;
    cursor: pointer;
    transition: .2s;

    &:hover {
      background: #ff6262;
    }
  }
}
