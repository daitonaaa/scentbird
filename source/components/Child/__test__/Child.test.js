import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Child } from 'components/Child/Child';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });


const defaultProps = {
  id: 1,
  check: false,
  title: 'something',
  parentId: 1,

  setCheckChildStatus: ()=>{},
  deleteItemChildAndSetCount: ()=>{},
};


describe('> > > COMPONENT - Child', () => {
  it('#create snapshot', () => {
    const output = shallow(
      <Child {...defaultProps} />
    );

    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('#test render check text', () => {
    const output = shallow(
      <Child {...defaultProps} />
    );

    expect(output.find('.child__check-btn').text()).toBe('Выделить');
  });

  it('#test method: handleDeleteItemChild => delete item', () => {
    const output = mount(
      <Child {...defaultProps} />
    );

    const wrapper = output.instance();
    const spy = jest.spyOn(wrapper, 'handleDeleteItemChild');
    wrapper.forceUpdate();
    output.find('.child__delete-btn').simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('#test method: handleToggleCheck => toggle check status', () => {
    const output = mount(
      <Child {...defaultProps} />
    );

    const wrapper = output.instance();
    const spy = jest.spyOn(wrapper, 'handleToggleCheck');
    wrapper.forceUpdate();
    output.find('.child__check-btn').simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
