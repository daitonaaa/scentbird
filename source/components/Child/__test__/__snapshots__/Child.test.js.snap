// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`> > > COMPONENT - Child #create snapshot 1`] = `
<div
  className="child"
>
  <div
    className="child__title"
  >
    <span>
      1
      :
    </span>
     
    something
  </div>
  <div
    className="child__check-btn"
    onClick={[Function]}
  >
    Выделить
  </div>
  <div
    className="child__delete-btn"
    onClick={[Function]}
  >
    Удалить
  </div>
</div>
`;
